### DISCUSSION TITLE
Developing formulae or casks

### DISCUSSION THREAD

**Discussion by ArkieSoft** [Homebrew/discussions#5096] [Writing Formulae/Casks]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.2.6-5-g2cb8efc
ORIGIN: https://github.com/Homebrew/brew
HEAD: 2cb8efc51d1bf30adb71004d6921ca8beaae9994
Last commit: 21 hours ago
Core tap HEAD: e6e5e758686f1d7a6969845efdb32aafe194769f
Core tap last commit: 2 weeks ago
Core tap branch: shurl
Core tap JSON: 30 Jan 13:08 UTC
Core cask tap HEAD: 5f1cfce8225f922e4b5420184c7171152702c2a6
Core cask tap last commit: 53 minutes ago
Core cask tap JSON: 30 Jan 13:08 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: nvim
HOMEBREW_MAKE_JOBS: 8
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 3.1.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.43.0 => /opt/homebrew/bin/git
Curl: 8.4.0 => /usr/bin/curl
macOS: 14.2.1-arm64
CLT: 15.1.0.0.1.1700200546
Xcode: 15.2
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: You have uncommitted modifications to Homebrew/homebrew-core.
If this is a surprise to you, then you should stash these modifications.
Stashing returns Homebrew to a pristine state but can be undone
should you later need to do so for some reason.
  cd /opt/homebrew/Library/Taps/homebrew/homebrew-core && git stash -u && git clean -d -f

Uncommitted files:
   M Formula/s/shurl.rb

Warning: Some taps are not on the default git origin branch and may not receive
updates. If this is a surprise to you, check out the default branch with:
  git -C $(brew --repo homebrew/core) checkout master
```

### Description of issue

So I am trying to add my wrapper to brew but when I go through the test, it pops up this error 

>Fetching gem metadata from https://rubygems.org/.......
>Fetching concurrent-ruby 1.2.3
Fetching minitest 5.21.2
Fetching sorbet-runtime 0.5.11219
Installing sorbet-runtime 0.5.11219
Installing minitest 5.21.2
Installing concurrent-ruby 1.2.3
Installing activesupport 6.1.7.6
Bundle complete! 36 Gemfile dependencies, 16 gems now installed.
Bundled gems are installed into `../../../opt/homebrew/Library/Homebrew/vendor/bundle`
Removing concurrent-ruby (1.2.2)
Removing minitest (5.21.1)
Error: Testing requires the latest version of shurl

But there is only one version of my package. I am probably missing something or maybe I need to re-release my package (Which I would rather no do, but will if needed). Any help is very much appreaciated.

Link to shurl.rb https://github.com/ArkieSoft/Shurl-cli/blob/v1.0/shurl.rb

**Comment by SMillerDev** [comment]
What does `brew info shurl` say? I would guess you don't have it installed.

---
> So I am trying to add my wrapper to brew but when I go through the test, it pops up this error

If you want to submit it to Homebrew/core, please note: https://docs.brew.sh/Acceptable-Formulae#niche-or-self-submitted-stuff
